### PR TITLE
Update for use in part 1 step 4

### PR DESCRIPTION
**Pull request by mmschaefer**
Updating to latest requirement

### CODE REVIEW

File: src/org/etools/j1939_84/model/SpnGroup.java [hunk 1]
```diff
@@ -23,6 +24,15 @@ public boolean isSatisfied(Collection<Integer> supportedSpnValues) {
         return supportedSpnValues.stream().anyMatch(spns::contains);
     }
 
+    public boolean containsMultiple(Collection<Integer> supportedSpnValues) {
+        Collection<Integer> matchingSpns = new ArrayList<>();
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/model/SpnGroup.java:28]: this variable isn't used.

File: src/org/etools/j1939_84/model/SpnGroup.java [hunk 2]
```diff
@@ -23,6 +24,15 @@ public boolean isSatisfied(Collection<Integer> supportedSpnValues) {
...
+        if (spns.isEmpty()) {
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/model/SpnGroup.java:29]: You don't need to even check for empty.  The logic below will return false if `spns` is empty.

File: src/org/etools/j1939_84/modules/SupportedSpnModule.java [hunk 3]
```diff
@@ -341,4 +366,43 @@ public boolean validateFreezeFrameSpns(ResultsListener listener, Collection<Inte
         return result;
     }
 
+    private static Collection<SpnGroup> getNoticedNoDupDataStreamSpns(FuelType fuelType, int engineModelYear) {
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/modules/SupportedSpnModule.java:369]: `fuelType` isn't used.

File: src/org/etools/j1939_84/modules/SupportedSpnModule.java [hunk 4]
```diff
@@ -341,4 +366,43 @@ public boolean validateFreezeFrameSpns(ResultsListener listener, Collection<Inte
...
+        Collection<SpnGroup> requiredSpns = new HashSet<>(NOTICED_NON_DUP_DATA_STREAM_SPNS);
+
+        if (engineModelYear >= 2024) {
+            requiredSpns.addAll(NOTICED_NON_DUP_2024_DATA_STREAM_SPNS);
+        }
+
+        return requiredSpns.stream().sorted().collect(Collectors.toList());
+    }
+
+    private static Collection<SpnGroup> getDesiredNoDupDataStreamSpns(FuelType fuelType, int engineModelYear) {
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/modules/SupportedSpnModule.java:379]: Neither parameter is used here.

File: src/org/etools/j1939_84/model/SpnGroup.java [hunk 5]
```diff
@@ -23,6 +24,15 @@ public boolean isSatisfied(Collection<Integer> supportedSpnValues) {
         return supportedSpnValues.stream().anyMatch(spns::contains);
     }
 
+    public boolean containsMultiple(Collection<Integer> supportedSpnValues) {
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/model/SpnGroup.java:27]: This method should have a unit test.

File: src/org/etools/j1939_84/modules/SupportedSpnModule.java [hunk 6]
```diff
@@ -341,4 +366,43 @@ public boolean validateFreezeFrameSpns(ResultsListener listener, Collection<Inte
...
+        Collection<SpnGroup> requiredSpns = new HashSet<>(DESIRED_NON_DUP_DATA_STREAM_SPNS);
+
+        return requiredSpns.stream().sorted().collect(Collectors.toList());
+    }
+
+
+    public boolean isMoreThanOneSpnReportedInfo(ResultsListener listener, Collection<Integer> spns, FuelType fuelType, int engineModelYear){
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/modules/SupportedSpnModule.java:386]: These methods don't seem to be used and they are missing unit tests.

File: src/org/etools/j1939_84/model/SpnGroup.java (hunk 1 above)
Comment (mmschaefer) [reply src/org/etools/j1939_84/model/SpnGroup.java:28]: I am assuming you meant method not variable.  The method is used in the part 1 step 4 that is also up for review.

File: src/org/etools/j1939_84/modules/SupportedSpnModule.java (hunk 6 above)
Comment (mmschaefer) [reply src/org/etools/j1939_84/modules/SupportedSpnModule.java:386]: The code is coming that it is used in.

Comment (mmschaefer) [reply src/org/etools/j1939_84/modules/SupportedSpnModule.java:386]: It is used in the part 1 step 4 which is also up for review.  I have tickets in for the unit tests.

File: src/org/etools/j1939_84/modules/SupportedSpnModule.java (hunk 4 above)
Comment (mmschaefer) [reply src/org/etools/j1939_84/modules/SupportedSpnModule.java:379]: Parameters removed

**Pull request merged**